sematic/resolvers/log_streamer.py

```diff
@@ -201,39 +300,56 @@ def ingested_logs(
         # true when the problem is something with the "normal" logging mechanisms, like
         # a failure to upload the logs to remote. Having *some* way to see what the code
         # was doing before it died will be essential.
-        _tail_log_file(file_path, max_tail_bytes)
+        _tail_log_file(file_path)
+        if read_file_descriptor is not None:
+            os.close(read_file_descriptor)
+        if write_file_descriptor is not None:
+            os.close(write_file_descriptor)
 
 
-def _tail_log_file(file_path, max_tail_bytes, print_func=None):
-    """Print the last lines of the log file.
+def _send_signal_or_kill(pid: int, signal_num: int, timeout_seconds: int):
+    """Send the signal to the given pid. If not exited by timeout, send SIGKILL
 
-    The code will quickly traverse to the correct file location rather than reading
-    through the whole log.
+    Parameters
+    ----------
+    pid:
+        The pid of the process to kill
+    signal_num:
+        The initial signal to send. Will be sent repeatedly until the process
```

Comment:
It's so that if the signal is sent before the child even has a chance to register a handler, it will still have a chance to receive the signal after it has registered said handler. Updated the docstring